tf_agents/cogment_verse_tf_agents/reinforce/reinforce.py

```diff
@@ -30,48 +31,47 @@ class ReinforceAgent:
     # pylint: disable=too-many-instance-attributes
     def __init__(
         self,
-        id,
-        obs_dim,
-        act_dim,
-        epsilon=0.01,
-        gamma=0.95,
+        id="*",
+        obs_dim=4,
+        act_dim=2,
+        gamma=0.99,
         lr=3e-4,
         max_replay_buffer_size=50000,
-        seed=42,
-        epsilon_schedule=None,
+        seed=21,
         lr_schedule=None,
+        model_params=None
     ):
-        self.id = id
+
         self._params = {}
+        self._params["id"] = id
         self._params["obs_dim"] = obs_dim
         self._params["act_dim"] = act_dim
         self._params["seed"] = seed
         self._params["max_replay_buffer_size"] = max_replay_buffer_size
         self._params["gamma"] = gamma
+        self._params["lr"] = lr
+        self._params["_lr_schedule"] = lr_schedule
+        self._params["model_params"] = model_params
```

Comment:
IIRC the point of this `_params` dict was to be able to log all the hyperparameters of the algorithm simple. If you but a schedule here, it's no longer "loggable" anyway.

I'd say the role of this `_params` need to be reassessed in any case. Here It looks like it's the data structure that is saved/loaded. Maybe just limit it's usage to that. You can have a constructor that takes the params dict (directly `ReinforceAgent(params)` or though the spreading syntax `ReinforceAgent(**params)`) that is called by a static `load` function or in the adapter's _create function. And have `save` put everything into the params dict and save it.

Reply:
1. "Savable" vs "Loggable": done
2. ReinforceAgent(**params): done
3. Static load: I don't believe I need one as initializing the params is two lines of code, if I understand you correctly.